tests/components/emoncms/test_config_flow.py

```diff
@@ -19,6 +21,74 @@
 }
 
 
+@pytest.mark.parametrize(
+    ("url", "api_key"),
+    [
+        (USER_INPUT[CONF_URL], "regenerated_api_key"),
+        ("http://1.1.1.2", USER_INPUT[CONF_API_KEY]),
+    ],
+)
+async def test_reconfigure(
+    hass: HomeAssistant,
+    emoncms_client: AsyncMock,
+    url: str,
+    api_key: str,
+) -> None:
+    """Test reconfigure flow."""
+    new_input = {
+        CONF_URL: url,
+        CONF_API_KEY: api_key,
+    }
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=SENSOR_NAME,
+        data=new_input,
+        unique_id=UNIQUE_ID,
+    )
+    await setup_integration(hass, config_entry)
+    result = await config_entry.start_reconfigure_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        new_input,
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "reconfigure_successful"
+    entry = hass.config_entries.async_get_entry(config_entry.entry_id)
+    assert entry.data == new_input
+
+
+async def test_reconfigure_api_error(
+    hass: HomeAssistant,
+    emoncms_client: AsyncMock,
+) -> None:
+    """Test reconfigure flow with API error."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=SENSOR_NAME,
+        data=USER_INPUT,
+        unique_id=UNIQUE_ID,
+    )
+    await setup_integration(hass, config_entry)
+    emoncms_client.async_request.return_value = EMONCMS_FAILURE
+    result = await config_entry.start_reconfigure_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        USER_INPUT,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": "api_error"}
+    assert result["description_placeholders"]["details"] == "failure"
```

Comment:
can we finish this one?

Reply:
I thought it was complete. I will check what i am missing...

Reply:
done